uint8 dir_left
uint16 para_left
uint8 dir_right
uint16 para_right